public/js/auth.js

```
@@ -167,3 +167,364 @@ async function verifyUserBeforeAction(actionDescription = 'perform this action')
+    if (appContainer) appContainer.style.display = 'none';
+    if (loginModal) loginModal.style.display = 'flex';
+}
+
+async function handleLogout(e) {
+    e.preventDefault();
+
+    if (typeof firebase !== 'undefined' && firebase.auth) {
+        try {
+            await firebase.auth().signOut();
+            currentUser = null;
+            userLinks = [];
```

ESLint

[error] Read-only global 'userLinks' should not be modified.

(no-global-assign)